### PR TITLE
GIX-2180: Add accountIdentifier UserTokenData

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

To open the TransactionModal or ReceiveModal in the ICP tokens table, I need to know which account to use based on the row clicked.

I could use the title, but that is fragile because the title might change and not be the name of the account.

Instead, I added an optional field to UserTokenData: accounIdentifier.

# Changes

* New field `accountIdentifier` in `UserTokenData`.
* Populate the new field in `icpTokensListUser`.
* Populate the new field in `tokensListUserStore`. I changed the implementation to read all the universes accounts instead of the balance. I needed it to get the account identifier.
* Use function overloading in helper `sumAccounts`. Convenient to make sure the return is a `bigint` if there are accounts. I also changed the functionality slightly to return 0 when the list is empty. That made more sense to me.

# Tests

* Adapt the tests expectation for `icpTokensListUser`.
* Adapt the tests expectation for `tokensListUserStore`.
* Add tests `sumAccounts` edge cases.

# Todos

- [ ] Add entry to changelog (if necessary).

Not yet necessary.

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #4060** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/4060?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/4060?utm_source=stack-comment).

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/lib/derived/tokens-list-user.derived.ts [hunk 1]
```diff
@@ -48,15 +46,28 @@ const convertToUserTokenData = ({
...
     return {
       ...baseTokenData,
       balance: "loading",
       actions: [],
     };
   }
   const fee = TokenAmountV2.fromUlps({ amount: token.fee, token });
-  const balance = TokenAmountV2.fromUlps({ amount: balanceUlps, token });
+  // For ICP we show the sum of all ICP accounts.
+  // For other tokens we show the main account balance because subaccounts are not yet supported.
```
Comment (dskloetd) [inline frontend/src/lib/derived/tokens-list-user.derived.ts:60]: I would expect `sumAccounts` to work either way because for other tokens `accountsList` is just a list with a single element. Is that not the case?

Comment (lmuntaner) [reply frontend/src/lib/derived/tokens-list-user.derived.ts:60]: Yes, that is the case, but I wanted to be explicit about what's the balance in the other rows.

At the moment, both would calculate the same balance, but on the other tokens, we don't support subaccounts and this lines represent the Main account, not the balance of all the accounts.

File: frontend/src/lib/derived/tokens-list-user.derived.ts [hunk 2]
```diff
@@ -48,15 +46,28 @@ const convertToUserTokenData = ({
...
+  const balance = TokenAmountV2.fromUlps({
+    amount: isUniverseNns(baseTokenData.universeId)
+      ? sumAccounts(accountsList)
+      : mainAccount.balanceUlps,
+    token,
+  });
+  // For ICP, the row represents all the ICP accounts. Therefore, we don't want to set on accountIdentifier.
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/derived/tokens-list-user.derived.ts]: yes, thanks. I changed it to "the", sounds even better

File: frontend/src/lib/derived/tokens-list-user.derived.ts [hunk 3]
```diff
@@ -67,22 +78,28 @@ const convertToUserTokenData = ({
...
   UserToken[]
 >(
-  [tokensListBaseStore, universesAccountsBalance, tokensStore, authStore],
+  [tokensListBaseStore, universesAccountsStore, tokensStore, authStore],
   ([tokensList, balances, tokens, authData]) =>
     tokensList.map((baseTokenData) =>
-      convertToUserTokenData({ baseTokenData, balances, tokens, authData })
+      convertToUserTokenData({
+        baseTokenData,
+        accounts: balances,
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/derived/tokens-list-user.derived.ts]: oops, I forgot to rename. Done

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

File: frontend/src/lib/derived/tokens-list-user.derived.ts (hunk 1 above)
Comment (dskloetd) [reply frontend/src/lib/derived/tokens-list-user.derived.ts:60]: But if we ever do support subaccounts for other tokens, then in all likelihood we want want their rows to show the balances of all their accounts as well, right?

For me, adding this unnecessary conditional logic just adds confusion.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/derived/tokens-list-user.derived.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/derived/tokens-list-user.derived.ts:60]: We will need to change the actions if we support subaccounts (depending on how we support them). The actions are for one wallet. If we support subaccounts, the actions will then be like ICP, a "GoToDetail" icon.

Or maybe we'll add all main and subaccounts in this same table. At the moment, the meaning is Main account, that's how I understand it and that's what I want the code to do.

Comment (dskloetd) [reply frontend/src/lib/derived/tokens-list-user.derived.ts:60]: But that is also what the code will do if you don't add all the extra logic.

Comment (lmuntaner) [reply frontend/src/lib/derived/tokens-list-user.derived.ts:60]: Yes, but when I read it it won't be clear what the intention of the code was and whether it was a bug as a feature or on purpose.
